highfidelity/hifi
# New Default Strafe Animations Plus Animation Tuning Support

This pr implements changes to the default avatar animation system.
The tasks addressed are:

1) https://highfidelity.manuscript.com/f/cases/17100
Add properties to MyAvatar for sprintSpeed and walkBackwardSpeed.  Also the speed of movement follows an eliptical function in the forward direction.  (x^2)/a + (y^2)/b = 1 where 'a' is the square of the walkBackwardSpeed and 'b' is the square of the forwardSpeed.  Making 'b' larger than 'a' means that the movement speed is greater in the forward direction.

2) https://highfidelity.manuscript.com/f/cases/17099
Doug added new animations for Strafe Left and Strafe Right to avatar-animation.json.  These look more appropriate for the desktop locomotion speeds.  Also added were HMD strafe animations that keep the pelvis facing forward when moving laterally to prevent conflict with the ik system that is making the hips follow the head orientation.   Walk forward was also improved.   Idle Settle was added for smoother transitions to idle.  And the characteristic speeds were changed to match the speeds at which the motions were authored.

3) https://highfidelity.manuscript.com/f/cases/17098
Made SnapShotPrev the default for motion blending.  This means that we now compute the destination pose each frame.  This prevents a jump at the end of the transition.

Also, this pr implements the printing of the stack of animations currently playing and their respective weights to the HUD stats.  Currently this stack includes linear blends for the hands, motion blends for movement and state machine blends such as between idle talk and idle stand.  

Characteristic speeds for the motion blends that can be set in the avatar-animation.json are now used to compute the alphas for move blends.  Previously these were hard coded in rig.cpp but they have been removed.

### Test Plan:

1. Open interface in desktop mode and press '3' for third person.
2. Press the '/' key to display the HUD statistics.
3. Press the forward, back, shift left and shift right arrow keys.  You should see the Speed: in the HUD statistic increase to the same value (around 2.6) regardless of your direction of travel.
4. If you press shift + an arrow key, this speed should triple in size.
5. Bring up the console ctrl-alt-j,  then set MyAvatar.walkSpeed = 5.
6. This should result in the max forward speed being around 5, while backwards and left and right should still be 2.5
7. Set MyAvatar.sprintSpeed = 1.5, and shift + forward should result in the speed being around 7.5
8. Set MyAvatar.walkBackwardSpeed = .5 and the back arrow should result in a speed slightly less than .5, while the forward speed should still be around 5
9. Now press the shift left arrow key and see your avatar side step.  If you set MyAvatar.walkBackwardSpeed = 2.5, then press shift left you should now see a walk to the left
10.  Put on the hmd go to 3rd person and take a step to the left you should see your avatar side step to the left.
